theme/color.go

```diff
@@ -0,0 +1,40 @@
+package theme
+
+import (
+	"image/color"
+)
+
+// Brighten takes a color and brightens / dims it by the given
+// amount. Each value in the amount represents a 10% increase
+// in the intensity of the color.  Negative values allow for
+// dimming the color. This is used to compute a color such as
+// a hightlight on hover based on a base color.
+func Brighten(c color.Color, amount float64) color.Color {
+	r, g, b, a := c.RGBA()
+	factor := 1.0 + (amount / 10.0)
+	fn := func(i uint32, f float64) uint8 {
+		return uint8(float64(i) * f / 255)
+	}
+	return color.RGBA{fn(r, factor), fn(g, factor), fn(b, factor), uint8(a)}
+}
+
+// PrimaryTextColor returns either {black, white} depending on
```

Comment:
Rather than comparing with Primary, shouldn't this compare the contrast ratio against the Background color - it is more likely that text is rendered over the background colour than the primary colour.

It also feels like we shouldn't need to check this each time, shouldn't each theme have a text colour that doesn't change?

Reply:
Agree - that it should be part of the theme and not computed on the fly.  Thats what I did originally in this PR, but that requires adding entries to the interface, which in turn introduces a breaking change.

Reply:
keep in mind that this is used only to compute the color of text that is rendered over a primary background.

It doesnt affect text that is rendered over normal backgrounds.